analyzers/src/SonarAnalyzer.CSharp/Rules/LossOfFractionInDivision.cs

```diff
@@ -106,6 +110,33 @@ private static bool TryGetTypeFromAssignmentToFloatType(SyntaxNode division, Sem
                 return type.IsAny(KnownType.NonIntegralNumbers);
             }
 
+            if (division.Ancestors().LastOrDefault(x => TupleExpressionSyntaxWrapper.IsInstance(x)) != null)
+            {
+                if (division.Ancestors().OfType<AssignmentExpressionSyntax>().FirstOrDefault() is { } assignmentSyntax
+                    && assignmentSyntax.MapAssignmentArguments() is { } assignmentMappings)
+                {
+                    var assignementLeft = assignmentMappings.Where(x => x.Right.Equals(division)).FirstOrDefault().Left;
+                    if (assignementLeft != null)
+                    {
+                        type = semanticModel.GetTypeInfo(assignementLeft).Type;
+                        return type.IsAny(KnownType.NonIntegralNumbers);
+                    }
+                }
+                else if (division.Ancestors().OfType<VariableDeclarationSyntax>().FirstOrDefault() is { } variableDeclaration)
+                {
+                    var tuple = division.Ancestors().LastOrDefault(x => TupleExpressionSyntaxWrapper.IsInstance(x));
+                    var tupleArguments = ((TupleExpressionSyntaxWrapper)tuple).AllArguments();
+                    var declarationType = semanticModel.GetTypeInfo(variableDeclaration.Type).Type;
+                    List<ITypeSymbol> flattenTupleTypes = new();
+                    FlattenTupleType(declarationType, flattenTupleTypes);
+                    if (flattenTupleTypes.Any() && DivisionArgumentIndex(tupleArguments, division) is { } argumentIndex)
+                    {
+                        type = flattenTupleTypes[argumentIndex];
+                        return type.IsAny(KnownType.NonIntegralNumbers);
+                    }
+                }
+            }
+
             if (division.Parent.Parent.Parent is VariableDeclarationSyntax variableDecl)
```

Comment:
```suggestion
            if (division is { Parent: EqualsValueClauseSyntax { Parent: VariableDeclaratorSyntax { Parent: VariableDeclarationSyntax variableDecl } } })
```

Reply:
IMO pattern matching here introduces quite some noise visually.
I understand that if the first pattern, for example, does not match, there's an early return (right?), but the tradeoff of more readable code to this optimization should not be too great. 
Or am I missing something?

Reply:
This is more about the expected parent structure and readability. `division.Parent.Parent.Parent is VariableDeclarationSyntax` is just confusing if you don't see the intermediate steps.

Reply:
Ok, I see this. Thanks!